workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/backstack/ViewStateCache.kt

```diff
@@ -50,9 +51,14 @@ internal constructor(
       }
     },
     onRestored = { aggregator ->
-      currentOwner?.let { owner ->
-        aggregator.restoreRegistryControllerIfReady(owner.key, owner.controller)
-      }
+      currentOwner
+        // We're only allowed to restore from an INITIALIZED state, but this callback can also be
+        // invoked while the owner is already CREATED.
+        // https://github.com/square/workflow-kotlin/issues/570
+        ?.takeIf { it.lifecycle.currentState == Lifecycle.State.INITIALIZED }
```

Comment:
I think @rjrjr originally tried this solution but decided against it? I can't remember now.

Reply:
I believe it was just a matter of being able to reproduce the bug first.

If this line is removed, everything still passes except for the new test, even if you call `scenario.recreate()`, or change orientation or something.

Reply:
Yup, I feel fine about the fix, just didn't want to do it w/o proof.